requirements.txt

```diff
@@ -1,3 +1,3 @@
 fonttools==3.28.0
-ufoLib==2.1.1
+ufoLib==2.2.1
```

Comment:
if you're using features that are only available on `ufoLib>=2.2.1` then you need to also bump the minimum `install_requires` in the setup.py

Reply:
done, I was unaware it was needed in multiple places...

Reply:
yeah, the setup.py install_requires is what pip uses to resolve dependencies when one downloads a package from PyPI. The requirements.txt is just a list of packages, usually with pinned versions (`==1.2.3`), that can be used to create (virtual) environments in a reproducible way. It is only used in here to create the environment in which the test suite is run.

Reply:
https://caremad.io/posts/2013/07/setup-vs-requirement/